### PR TITLE
Audi UI

### PR DESCRIPTION
**Pull request by dreyhiflden**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        <p class="card__date">30.02.2025</p>
+      </div>
+      <div class="card a4">
+        <h3 class="headline headline--h3 headline--big">A4</h3>
+        <p class="card__description">Your Model</p>
+      </div>
+    </div>
+  </div>
+</body>
+</html>
```
Comment (carrot-e) [inline index.html]: No newline.

File: src/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
+.btn {
+  font-size: $base-font-size;
+  text-decoration: none;
+  border: 1px solid #000;
+  padding: 20px 10px;
```
Comment (carrot-e) [inline src/_buttons.scss]: Maybe move `20px`, `10px` to variables?
Same about border width.

File: src/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  display: inline-block;
+  vertical-align: middle;
+  font-family: $base-font-family;
+  text-align: center;
+  background: transparent;
+  min-width: calc(#{$base-font-size} * 9);
```
Comment (carrot-e) [inline src/_buttons.scss]: You can let Sass calculate this.
`min-width: $base-font-size * 9;`

File: src/_buttons.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      transition: transform 0.1s linear;
+    }
+    &:hover::after {
+      transform: translateX(8px);
+    }
+  }
+
+  &--text {
+    border: none;
+    color: #fff;
```
Comment (carrot-e) [inline src/_buttons.scss]: I'm not sure, every .btn--text will be white. Can you persuade me?

File: src/_card.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.card {
+  display: grid;
+  padding: 20px 20px;
+  font-size: $base-font-size;
+  font-family: $base-font-family;
+  background: #fff;
+  grid-template-rows: max-content max-content auto;
+
+  &--colored {
+    background: #BB0A30;
```
Comment (carrot-e) [inline src/_card.scss]: I'd create a variable for this color.

File: src/_card.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    color: #fff;
+  }
+
+  &--bg-image {
+    background: url("../img/card-390x480.jpg") center/cover no-repeat;
```
Comment (carrot-e) [inline src/_card.scss]: According to this style, every card--bg-image will use this image as a background. Is it the intended behavior?

File: src/_card.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    color: #fff;
+  }
+
+  &__date {
+    font-size: 14px;
```
Comment (carrot-e) [inline src/_card.scss]: Can this be declared in typography components?

File: src/_typography.scss [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+  &--h1 {
+   font-weight: 800;
+   font-size: 46px;
+   line-height: 52px;
+  }
+
+  &--h2 {
+    font-weight: 800;
+    font-size: 46px;
```
Comment (carrot-e) [inline src/_typography.scss]: --h2 is equal to --h1, isn't it? Why?

File: src/_typography.scss [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+    line-height: 52px;
+  }
+
+  &--calendar {
```
Comment (carrot-e) [inline src/_typography.scss]: I would prefer a more abstract modifier.
Like --xl or smth.

File: src/_typography.scss [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+  font-weight: 300;
+
+  &--shortread {
+    line-height: 1.5;
+  }
+
+  &--small {
+    line-height: 1.5;
+    opacity: 40%;
+    font-size: calc(#{$base-font-size} - 2px)
```
Comment (carrot-e) [inline src/_typography.scss]: Can be done with Sass, too

File: src/_card.scss (hunk 6 above)
Comment (dreyhiflden) [reply src/_card.scss]: In this case, we have only one card width image as background, so I thinks it's ok to use this generic class name. If we need to use more images as background for other cards, we need to create more specific classes for every separate background image.

File: src/_buttons.scss (hunk 4 above)
Comment (dreyhiflden) [reply src/_buttons.scss]: Agree, text buttons not only have the white color. I created specific modifier to set text color for the button.

File: src/_card.scss (hunk 7 above)
Comment (dreyhiflden) [reply src/_card.scss]: Yes. Changed.

File: src/_typography.scss (hunk 8 above)
Comment (dreyhiflden) [reply src/_typography.scss]: Fixed.

File: src/_typography.scss (hunk 9 above)
Comment (dreyhiflden) [reply src/_typography.scss]: Done.

File: src/_typography.scss (hunk 10 above)
Comment (dreyhiflden) [reply src/_typography.scss]: Changed.

File: index.html (hunk 1 above)
Comment (dreyhiflden) [reply index.html]: Fixed.

File: src/_buttons.scss (hunk 3 above)
Comment (dreyhiflden) [reply src/_buttons.scss]: Changed.

File: src/_buttons.scss (hunk 2 above)
Comment (dreyhiflden) [reply src/_buttons.scss]: Changed.

File: src/_variables.scss [hunk 11]
```diff
@@ -3,3 +3,8 @@ $base-font-size: 16px;
 
 $tablet: 480px;
 $desktop: 1024px;
+
+$btn-papping: 20px 10px;
+$btn-border: 1px solid #000;
+
+$card-colored: #BB0A30;
```
Comment (carrot-e) [inline src/_variables.scss]: No newline

File: src/_card.scss (hunk 6 above)
Comment (carrot-e) [reply src/_card.scss]: Generally speaking, I agree with your point.

But the idea of the task was to:
* create _generic_ UI components
* use them to build _specific_ layout.

In this case, you are defining specific image for a generic UI component which doesn't seem right.
 
> create more specific classes for every separate background image

This solution sounds reasonable, why not implement it?

Comment (dreyhiflden) [reply src/_card.scss]: Now I rethought this problem and rewrite the `card - bg-image` modification into the` card__bg-image` element and created a special `old-man-photo` modification to set these images as background. As my point of view, this is the best decision.

File: src/_card.scss (hunk 5 above)
Comment (dreyhiflden) [reply src/_card.scss]: Done.

File: src/_card.scss [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+.old-man-bg {
+  background: url("../img/card-390x480.jpg") center/cover no-repeat;
+}
+
+@media all and (min-width: $tablet) {
+
+}
+
+@media all and (min-width: $desktop) {
+    .card {
```
[the remark this answers is not in the document]
Comment (dreyhiflden) [reply src/_card.scss]: Fidex

**Pull request merged**